crates/wysiwyg/src/dom/dom_methods.rs

```diff
@@ -63,7 +63,7 @@ where
 
     pub fn replace_text_in(&mut self, new_text: S, start: usize, end: usize) {
         #[cfg(any(test, feature = "assert-invariants"))]
-        self.assert_invariants();
+        self.start_transaction();
```

Comment:
By starting the transaction here, we're saying that `replace_text_in()` cannot itself be composed within a transaction. Was that the intention or should the transaction be moved into the composer layer?

Reply:
Oh yeah I haven't thought about that directly. I was mostly thinking about "let's do most of the transactions code inside the Dom layer", but composition might be an issue for this kind of functions. 
Maybe we need to think a bit about what we want to achieve. I think I'd like a solution were a dom method can perform a transaction with multiple other methods, but be available as well for composition 😅 

Maybe we shouldn't fail if a transaction is already in progress, and make it possible to compose transaction with other transaction, but I guess we would have to modify the mechanism a bit (like, attach an id or something to a transaction, and make it so the transaction can only be ended by its initiator) 🤔

Reply:
The thinking behind disallowing nested transactions was that it would simplify things and encourage us to design Dom methods that should usually preserve the invariants. But equally I can see how it might be an unnecessary constraint.

> we would have to modify the mechanism a bit (like, attach an id or something to a transaction, and make it so the transaction can only be ended by its initiator

Yes I think so - I wonder if a macro would help us here (maybe automatically assign the caller name as an ID and ensure the transaction is ended at the end of a method)?

Reply:
BTW I've been checking on this and actually there is no direct way to get the caller function name / id: https://github.com/rust-lang/rfcs/pull/2818

This is pretty annoying because I guess we'd have to explicitly pass some information to the caller so it can end the transaction, which makes the API less nice to use.